src/graphhopper/router.js

```diff
@@ -1,9 +1,50 @@
 import express from 'express';
 import graphHopperClient from './client.js';
 import logger from '../lib/logger.js';
+import * as gtfsRtClient from '../gtfs-rt/client.js';
+import { mergeAlertsIntoRoutes } from '../gtfs-rt/alerts.js';
 
 const router = express.Router();
-const routes = [
+router.use((req, res, next) => {
+  // you only want to cache for GET requests
+  if (req.method == 'GET') {
+    res.set('Cache-Control', 'public, max-age=120');
+  } else {
+    // for the other requests set strict no caching parameters
+    res.set('Cache-Control', `no-store`);
+  }
+  next();
+});
+
+router.get('/route-pt', async (req, res) => {
+  try {
+    const alertPromise = gtfsRtClient.getAlerts();
+    const graphHopperPromise = graphHopperClient.request({
+      method: 'get',
+      url: req.url,
+    });
+
+    const [alertResult, graphHopperResult] = await Promise.allSettled([
+      alertPromise, graphHopperPromise
+    ]);
+
+    if (graphHopperResult.status === 'rejected') throw graphHopperResult.reason;
+    if (alertResult.status === 'rejected') logger.error(alertResult.reason);
+
+    res.send(mergeAlertsIntoRoutes(alertResult.value, graphHopperResult.value.data));
+  } catch (error) {
+    logger.error(error);
+    if (error.response) {
+      res.status(error.response.status).send(error.response.data);
+    }
+    else {
+      res.status(500);
+    }
+    res.end();
```

Comment:
move this to a `finally` block. as in:
```javascript
try {...}
catch {...}
finally { res.end() }
```

Reply:
Same question as above -- why is that better? What's the difference? As above, this usage is unchanged from the code on main that you wrote.